client/session_write_test.go

```diff
@@ -242,16 +219,28 @@ func testWriteConsistencyLevel(
 
 	// Callback
 	enqueueWg.Wait()
+	host := session.topoMap.Hosts()[0] // any host
 	writeErr := "a specific write error"
-	for i := 0; i < session.topoMap.Replicas()-failures; i++ {
-		completionFn(nil, nil)
+	for i := 0; i < success; i++ {
+		completionFn(host, nil)
 	}
 	for i := 0; i < failures; i++ {
-		completionFn(nil, fmt.Errorf(writeErr))
+		completionFn(host, fmt.Errorf(writeErr))
 	}
 
-	// Wait for write to complete
-	writeWg.Wait()
+	// Wait for write to complete or timeout
+	doneCh := make(chan struct{})
+	go func() {
+		writeWg.Wait()
+		close(doneCh)
+	}()
+
+	select { // NB(bl): Check that we're correctly signaling in write_state.completionFn.
```

Comment:
If the completionFn is incorrect, the write above will hang. Re-worded for clarity.